### DISCUSSION TITLE
How to customize 'Go To Definition' behavior

### DISCUSSION THREAD

**Discussion by c3-WesleyPickering** [microsoft/pylance-release#3595] [Q&A]
We have a unique use case where we can define a Python class in a non .py file and have the functions for the class be implemented in a .py file with the same name as that class.

Ex. `Racecar.ext` define a class and function signatures and Racecar.py has the implementation for those functions.

To use Pylance's IntellSense functionalities, we have a folder of pyi files for all the classes in the stubs folder of the extension, and an `__init__.pyi` file to declare the class imports. This provides us with autocomplete.

We are also looking to customize the behavior of `Go To Definition` so that:
- When we call it on the class name it should jump to the our custom Class definition file (Racecar.ext).
- When we call it on a class method it should jump to the implementation file (Racecar.py)

**Comment by rchiodo** [comment]
I think this would require changes to the import resolver:
https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/analyzer/importResolver.ts

It only looks for py and pyi files for supported files.
https://github.com/microsoft/pyright/blob/ff43f7f0939dc00c17cd1c8bc3b284e6e3c071c5/packages/pyright-internal/src/analyzer/importResolver.ts#L100

And that would only work if your '.ext' files were valid python syntax.

You could also provide your own python language server that supports goto definition only. VS code would combine your results with ours.
